database/sqlite/sqlite_functions.c

```diff
@@ -1270,9 +1273,9 @@ RRDHOST *sql_create_host_by_uuid(char *hostname)
 
     host->system_info = callocz(1, sizeof(*host->system_info));;
     rrdhost_flag_set(host, RRDHOST_FLAG_ARCHIVED);
-#ifdef ENABLE_DBENGINE
-    host->rrdeng_ctx = &multidb_ctx;
-#endif
+
+    // Create multidb engine instance if necessary
+    host->rrdeng_ctx = storage_engine_new(storage_engine_get(RRD_MEMORY_MODE_DBENGINE), host);
```

Comment:
Configure to compile without dbengine `--disable-dbengine`.

database/sqlite/sqlite_functions.c: In function 'sql_create_host_by_uuid':
database/sqlite/sqlite_functions.c:1278:24: warning: implicit declaration of function 'storage_engine_new' [-Wimplicit-function-declaration]
 1278 |     host->rrdeng_ctx = storage_engine_new(storage_engine_get(RRD_MEMORY_MODE_DBENGINE), host);
      |                        ^~~~~~~~~~~~~~~~~~
database/sqlite/sqlite_functions.c:1278:43: warning: implicit declaration of function 'storage_engine_get' [-Wimplicit-function-declaration]
 1278 |     host->rrdeng_ctx = storage_engine_new(storage_engine_get(RRD_MEMORY_MODE_DBENGINE), host);
      |                                           ^~~~~~~~~~~~~~~~~~
database/sqlite/sqlite_functions.c:1278:22: warning: assignment to 'STORAGE_ENGINE_INSTANCE *' {aka 'struct storage_engine_instance *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
 1278 |     host->rrdeng_ctx = storage_engine_new(storage_engine_get(RRD_MEMORY_MODE_DBENGINE), host);